### PR TITLE
feat..Detail page was added with author information and comments section

### PR DESCRIPTION
**Pull request by BayramovaKonul**
Url configuartions have been set up for each article's details, author info and comment section. Blog, categories and details pages are fully functional.

The search in the category page doesn't work as intended. Text in the detail page is unreadable.

### CODE REVIEW

File: blog/urls.py [hunk 1]
```diff
@@ -8,5 +8,5 @@
     path('contact', views.contact, name = 'contact'),
     path('blogs', views.blogs, name = 'blogs'),
     path('category/<slug:category_slug>', views.category_blog, name = 'category'),
-    path('blogs/<int:id>/', views.detail, name = 'blogs/<int:id>/'),
+    path('blogs/<int:blog_id>/', views.detail, name = 'blog/details'),
```
Comment (valiyevvali) [inline blog/urls.py]: dynamic part must be blog's slug not id. For SEO oriented urls

File: blog/views.py [hunk 2]
```diff
@@ -40,12 +40,34 @@ def blogs (request):
...
+        'category_name' : slug_categories.name,
+        'recent_posts' : recent_posts,
+        'categories' : categories
     })
 
-def detail (request,id):
-    return render(request, 'detail.html')
\ No newline at end of file
+
+def detail (request,blog_id):
```
Comment (valiyevvali) [inline blog/views.py]: too general function name

File: blog/views.py [hunk 3]
```diff
@@ -40,12 +40,34 @@ def blogs (request):
...
+    details = ArticleModel.objects.get(id = blog_id)
```
Comment (valiyevvali) [inline blog/views.py]: Check maybe blog with this id(replace it with slug) does not exist

File: blog/views.py [hunk 4]
```diff
@@ -40,12 +40,34 @@ def blogs (request):
...
+    comments = CommentModel.objects.filter(article_id = blog_id)
+    blogs = ArticleModel.objects.all()
+    recent_posts = blogs.order_by('-created_at')[:4]
+    categories = CategoryModel.objects.annotate(article_count=Count('articles')).order_by("-article_count").values('name', 'article_count', 'slug')
+    return render(request, 'detail.html', context={
```
Comment (valiyevvali) [inline blog/views.py]: template name (detail.html) is too common.Can be `blog_detail`

File: blog/views.py [hunk 5]
```diff
@@ -40,12 +40,34 @@ def blogs (request):
                  
 
 def category_blog (request, category_slug):
-    categories = get_object_or_404(CategoryModel, slug = category_slug)  # check the table to find asked category_slug
-    category_blogs = categories.articles.all() # get all articles from table based on the category using related name
+    categories = CategoryModel.objects.annotate(article_count=Count('articles')).order_by("-article_count").values('name', 'article_count', 'slug')
+    slug_categories = get_object_or_404(CategoryModel, slug = category_slug)  # check the table to find asked category_slug
```
Comment (valiyevvali) [inline blog/views.py]: naming is not correct, it is returning 1 category object so it should be singular. can be `category_obj`

**Comment by valiyevvali** [comment]
Remove __py_cache__ folders and add it to gitignore

File: blog/views.py (hunk 3 above)
Comment (BayramovaKonul) [reply blog/views.py]: we go to article_detail using its display on the page (such as articles on blog page). If it doesn't exist, how can it be shown on the page?

Comment (valiyevvali) [reply blog/views.py]: You are right, but we can manually write `articles/random_slug` in browser tab ,in that case your code will not work properly.
But If you handle this case , you can show 404 page to user which is correct option.

**Pull request merged**